modules/util/util.go

```diff
@@ -63,3 +64,52 @@ func Min(a, b int) int {
 func IsEmptyString(s string) bool {
 	return len(strings.TrimSpace(s)) == 0
 }
+
+// NormalizeEOL will convert Windows (CRLF) and Mac (CR) EOLs to UNIX (LF)
+func NormalizeEOL(input []byte) []byte {
+	var right, left, pos int
+	if right = bytes.IndexByte(input, '\r'); right == -1 {
+		return input
+	}
+	length := len(input)
+	tmp := make([]byte, length)
+	for left < length && input[left] == '\n' {
+		left++
+	}
+	if left != 0 {
+		copy(tmp[pos:pos+left], input[0:left])
+		pos = left
+		right -= left
+	}
+
+	if left < length {
+		if input[left] == '\n' {
+			left++
+			right--
+		}
+		copy(tmp[pos:pos+right], input[left:left+right])
+		pos += right
+		tmp[pos] = '\n'
+		left += right + 1
+		pos++
+	}
+
+	for left < length {
+		if input[left] == '\n' {
+			left++
+		}
+
+		right = bytes.IndexByte(input[left:], '\r')
```

Comment:
In Java, the JVM JIT has a weird feature whereby it's actually quicker to assign names used inside loops rather than refer to things used outside the loops, e.g. `right := ...` would be quicker than `right = ...`. On testing, this does not appear to be the case for go's compiler.